### PR TITLE
Add privacy considerations for cached/persisted data

### LINKED ISSUE #254: Privacy considerations for cached/persisted data
[Leaving feedback on behalf of Chrome privacy reviewers]

Does the API result in any data bring cached/persisted in the browser, or by the hardware accelerators? For example, do you have any compilation caches similarly to [WebGL/WebGPU shaders](https://gpuweb.github.io/gpuweb/#privacy-user-agent-state)?

- Any new cache/storage should be partitioned as per [storage partitioning](https://github.com/privacycg/storage-partitioning) work that is currently underway in browsers.
- Deletion operations (e.g. when a user clears site data in the browser) should be defined for such data.

### PR DESCRIPTION
**Pull request by anssiko**
Fix #254

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/pull/259.html" title="Last updated on Apr 4, 2022, 3:55 PM UTC (887eafc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/259/5b6913b...887eafc.html" title="Last updated on Apr 4, 2022, 3:55 PM UTC (887eafc)">Diff</a>

### CODE REVIEW

**Comment by anssiko** [comment]
(PR rebased, build passes.)

File: index.bs [hunk 1]
```diff
@@ -394,7 +394,9 @@ No information from the underlying platform is exposed directly. An execution ti
 
 Note: The group is <a href="https://github.com/webmachinelearning/webnn/issues/85">soliciting further input</a> on the proposed execution time analysis fingerprinting vector and will augment this section with more information and mitigations to inform the implementers of this API.
 
-Implementers of this API are expected to be familiar with the <a href="https://gpuweb.github.io/gpuweb/#security-privacy">WebGPU Privacy Considerations</a>.
+This API mitigates against timing attacks that rely on data caching or persistence by not intrinsically supporting custom shader authoring. Instead, the API builds upon pre-existing shaders and lower level primitives of the browser or the underlying OS. Web developers who interface with {{GPUDevice}} are expected to be aware of <a href="https://gpuweb.github.io/gpuweb/#privacy-user-agent-state">WebGPU compilation cache considerations</a>.
```
Comment (krgovind) [inline index.bs:397]: I think this sentence may be a bit backward. Should we say "Unlike APIs like WebGL, and WebGPU; this API does not intrinsically supporting custom shader authoring; and as a result is not prone to timing attacks that rely on shader caches, or other persistent data. "?

**Comment by anssiko** [comment]
Thanks @krgovind et al. for the suggestion. The PR has been updated, PTAL.

**Comment by anssiko** [comment]
@krgovind, gentle ping PTAL the updated PR (I'll resolve the merge conflict after getting your review.)

**Review by krgovind — APPROVED** [review]
Looks great! Thank you.

**Review by huningxin — APPROVED** [review]
LGTM, thanks @anssiko !

**Comment by anssiko** [comment]
With two reviews, I'll merge this. Thanks @krgovind and @huningxin!

**Pull request merged**